src/backend/marsha/core/api.py

```diff
@@ -673,24 +673,45 @@ def get_permissions(self):
         if self.action == "metadata":
             permission_classes = [permissions.IsVideoToken]
         elif self.action in ["create", "list"]:
-            # NB: list should be restricted by queryset, not resource id
             permission_classes = [
-                permissions.IsTokenInstructor | permissions.IsTokenAdmin
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
             ]
         else:
             permission_classes = [
                 permissions.IsTokenResourceRouteObjectRelatedVideo
                 & permissions.IsTokenInstructor
                 | permissions.IsTokenResourceRouteObjectRelatedVideo
                 & permissions.IsTokenAdmin
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
             ]
         return [permission() for permission in permission_classes]
 
     def get_queryset(self):
         """Restrict list access to timed text tracks related to the video in the JWT token."""
         user = self.request.user
-        if isinstance(user, TokenUser):
+        if isinstance(user, TokenUser) and not user.token.get("user_id"):
             return TimedTextTrack.objects.filter(video__id=user.id)
+
+        if isinstance(user, TokenUser) and user.token.get("user_id") == user.token.get(
```

Comment:
FYI @lunika @kernicPanel I discussed this with @mbenadda tonight and think we are facing a complexity challenge with these 2 types of JWT tokens that are using a wrong terminology (calling "UserToken" what is in fact a VideoToken...).

The mitigation actions I see:
- internalizing the DRF JWT lib to simplify it to our use case AND rename things properly. Otherwise the code will continue to get more and more confusing
- Make use of the username and/or email passed by LTI for teachers so that we can create user accounts on the fly the same way as it is done on [Ashley](https://github.com/openfun/ashley) (reusing our [LTI toolbox lib](https://github.com/openfun/django-lti-toolbox)). It's the only way to offer the same experience to the teacher on the site and on the LMS. Otherwise, the teachers will see their organizations and playlists on the site, but not in the LTI XBlock... this will be confusing. Also, the deeplinking feature is much more interesting if we recognize the teachers and give them access to their objects.

Reply:
> FYI @lunika @kernicPanel I discussed this with @mbenadda tonight and think we are facing a complexity challenge with these 2 types of JWT tokens that are using a wrong terminology (calling "UserToken" what is in fact a VideoToken...).
> 
> The mitigation actions I see:
> 
>     * internalizing the DRF JWT lib to simplify it to our use case AND rename things properly. Otherwise the code will continue to get more and more confusing
> 
>     * Make use of the username and/or email passed by LTI for teachers so that we can create user accounts on the fly the same way as it is done on [Ashley](https://github.com/openfun/ashley) (reusing our [LTI toolbox lib](https://github.com/openfun/django-lti-toolbox)). It's the only way to offer the same experience to the teacher on the site and on the LMS. Otherwise, the teachers will see their organizations and playlists on the site, but not in the LTI XBlock... this will be confusing. Also, the deeplinking feature is much more interesting if we recognize the teachers and give them access to their objects.

For now, I don't see clearly how we can manage login/password through classic LTI request, but I totally agree with you :100:

Reply:
:+1: can you open an issue on this subject ?

Reply:
done